### ISSUE TITLE
Unable to upload new files to a folder

### ISSUE THREAD

**Issue by bgrich** [CenterForOpenScience/osfr#48]
I've been unable to upload new files to a folder using `upload_files()`. It looks like the link constructed by `construct_link_files()` gives a direct WaterButler link to the projects base directory. To upload to a folder within the project we need the upload link for that particular folder. 

The fix I have been thinking of would be to check the user's provided file destination for a path that:

* Is empty, causing the file to be uploaded to the base directory with the local file name (currently implemented). 
* Has a file name, causing the file to be uploaded to the base directory with the provided file name (currently implemented). 
* Has a path including a folder and a file name, causing the file to be uploaded to the folder with the given file name. 
* Has a path with only a folder, causing the file to be uploaded to the folder with the local file name.

Since `upload_files()` already checks the given project id for file information, the upload link for the folder can be extracted from that information. I think that this would all be handled within `upload_new()`.

Additionally, this fix could also be applied to `create_folder()` which has a similar problem.

**Comment by hathawayj** [comment]
I am experiencing the same problem and would love to have this fixed as soon as possible.

**Comment by hathawayj** [comment]
I took a stab at fixing this.  It works for my needs. 

https://github.com/hathawayj/osfr/commit/4af3b2f03f492aedd7a62e00b97a0a8b96f2cc4f

I fixed the functions `create_folder`, `upload_files`, and `upload_new` in the the `files.R` and `folders.R` function files.  I rebuilt the help files in this commit as well.  

With the changes, we can create nested subfolders and save files to those nested subfolders.
